renku/core/management/clone.py

```diff
@@ -53,29 +53,51 @@ def clone(
         os.environ["GIT_LFS_SKIP_SMUDGE"] = "1"
 
     try:
-        repo = Repo.clone_from(url, path, recursive=recursive, depth=depth, progress=progress)
-    except GitCommandError as e:
-        if not raise_git_except:
-            lines = progress.other_lines + progress.error_lines if progress else []
-            error = "".join([f"\n\t{line}" for line in lines if line.strip()])
-            message = f"Cannot clone remote Renku project: Git exited with code {e.status} and error message:\n {error}"
-            raise errors.GitError(message)
+        kwargs = {}
+
+        if checkout_rev:
+            kwargs["branch"] = checkout_rev
 
-        raise e
+        repo = Repo.clone_from(url, path, recursive=recursive, depth=depth, progress=progress, **kwargs)
+    except GitCommandError as e:
+        # NOTE: clone without rev in case it was not a branch
+        if not checkout_rev:
+            if not raise_git_except:
+                lines = progress.other_lines + progress.error_lines if progress else []
+                error = "".join([f"\n\t{line}" for line in lines if line.strip()])
+                message = (
+                    f"Cannot clone remote Renku project: Git exited with code {e.status} and error message:\n {error}"
+                )
+                raise errors.GitError(message)
+
+            raise e
+        try:
+            repo = Repo.clone_from(url, path, recursive=recursive, depth=depth, progress=progress)
+        except GitCommandError as e:
+            if not raise_git_except:
+                lines = progress.other_lines + progress.error_lines if progress else []
+                error = "".join([f"\n\t{line}" for line in lines if line.strip()])
+                message = (
+                    f"Cannot clone remote Renku project: Git exited with code {e.status} and error message:\n {error}"
+                )
+                raise errors.GitError(message)
 
-    remote_refs = [Path(ref.abspath).name for ref in repo.remote().refs]
+            raise e
 
-    try:
-        if checkout_rev in remote_refs:
+        try:
```

Comment:
I find these repeated try/excepts a bit confusing - could you maybe add a few more inline comments to document the flow a bit better? Maybe something at the top of this whole block like:

1. try a clone assuming the `checkout_rev` is a branch
2. if this fails, try a clone without the branch
3. if this succeeds, try to checkout the `checkout_rev`